build-system/babel-plugins/babel-plugin-transform-inline-configure-component/index.js

```diff
@@ -0,0 +1,222 @@
+            )
+          );
+          path.parentPath.replaceWithMultiple(assignments);
+        }
+      },
+    },
+  };
+}
+
+/**
+ * Transforms using transformRedefineInline sub-plugin.
+ * @param {string} sourceFilename
+ * @param {Object} opts
+ * @return {Object}
+ */
+const redefineInline = (sourceFilename, opts) =>
+  transformFileSync(sourceFilename.toString(), {
+    configFile: false,
+    code: false,
+    ast: true,
+    sourceType: 'module',
+    plugins: [[transformRedefineInline, opts]],
+  });
+
+/**
+ * Replaces `configureComponent()` wrapping calls.
+ *
+ * @return {Object}
+ */
+module.exports = function({types: t}) {
+  function getImportPath(nodes, name) {
+    for (let i = 0; i < nodes.length; i++) {
+      const node = nodes[i];
+      if (
+        t.isImportDeclaration(node) &&
+        node.specifiers.find(({imported}) => t.isIdentifier(imported, {name}))
+      ) {
+        return node.source.value;
+      }
+    }
+  }
+
+  return {
+    name: 'transform-inline-decl-extensions',
+    visitor: {
+      CallExpression(path, {file}) {
+        if (!t.isIdentifier(path.node.callee, {name: calleeName})) {
+          return;
+        }
+
+        const [importedId, propsObj] = path.node.arguments;
+        if (!t.isIdentifier(importedId) || !t.isObjectExpression(propsObj)) {
+          return;
+        }
+
+        const program = path.findParent(p => t.isProgram(p));
+
+        const importPath = getImportPath(program.node.body, importedId.name);
+        if (!importPath) {
+          return;
+        }
+
+        for (const name in program.scope.bindings) {
+          if (name) {
+            path.scope.rename(name, program.scope.generateUid(name));
+          }
+        }
+
+        const propValues = Object.create(null);
+
+        for (const {key, value} of propsObj.properties) {
+          const {name} = key;
+
+          if (t.isMemberExpression(value)) {
+            throw path.buildCodeFrameError(
+              `${replacedMember} properties must not be assigned to members. ` +
+                'Set necessary values to program-level constants.'
+            );
+          }
+
+          if (t.isIdentifier(value)) {
+            const binding = path.scope.getBinding(value.name);
+            if (!binding || !t.isProgram(binding.scope.block)) {
+              throw path.buildCodeFrameError(
+                `ids used in ${replacedMember} must be defined as ` +
+                  'program-level constants.'
+              );
+            }
+            propValues[name] = value;
+            continue;
+          }
+
+          const id = program.scope.generateUidIdentifier(name);
+          program.scope.push({id, init: value, kind: 'const'});
+          propValues[name] = id;
+        }
+
+        const currentDirname = dirname(file.opts.filename);
+        const importedModule = join(currentDirname, importPath);
+
+        // TODO(alanorozco): sourcemaps
```

Comment:
Can you add a tracking issue for this, i believe you're possibly going to need resorcery and terser for this

Reply:
Added issue. Why `terser`? What am I missing :)

Reply:
https://github.com/ampproject/amphtml/pull/26779/files#diff-321517b76b9f18ac62dff6aa46ec79b1R102-R135

https://github.com/ampproject/amphtml/blob/master/build-system/compile/single-pass.js#L771-L803

i'm not sure but @kristoferbaxter and @jridgewell used it in these to fix up the source maps in conjunction with resorcery